xapian-bindings/android/CROSS_COMPILE

```diff
@@ -0,0 +1,121 @@
+Instructions for cross compiling for android 
+============================================
+This file is aimed to help developers get started with cross compiling xapian for android.
+
+
+Prerequisites
+=============
+1)
+You will need to download android ndk from https://developer.android.com/ndk/downloads/index.html 
+for your respective development platform. Extract it and now using the make-standalone-toolchain.sh
+in the ndk make a customized toolchain from the command line by doing the following:
+
+export NDK=<path_to_ndk>
+$NDK/build/tools/make-standalone-toolchain.sh --toolchain=<target-arch> --platform=<target-android-api> --install-dir=<path_to_ndk_standalone_toolchain>
```

Comment:
Trying to clean up our PRs, and looking into this one it seems this step is no longer needed - you can now just unzip the NDK and run the compiler from the resulting tree.  I've added an `android-cross` CI job which does this:

```
wget -Ondk.zip https://dl.google.com/android/repository/android-ndk-r25c-linux.zip             
unzip -q ndk.zip  
rm -f ndk.zip  
mv android-ndk-* ndk  
```
then:
```
PATH=$PWD/ndk/toolchains/llvm/prebuilt/linux-x86_64/bin:$PATH \  
    ./configure --disable-documentation \  
        --host armv7a-linux-androideabi33 \  
        CC=$PWD/ndk/toolchains/llvm/prebuilt/linux-x86_64/bin/armv7a-linux-androideabi33-clang \  
        CXX=$PWD/ndk/toolchains/llvm/prebuilt/linux-x86_64/bin/armv7a-linux-androideabi33-clang++ \  
        PKG_CONFIG=/bin/false  
```
and:
```
PATH=$PWD/ndk/toolchains/llvm/prebuilt/linux-x86_64/bin:$PATH make 
```

This seems to successfully build xapian-core.

I'll try to get the java bindings to build in CI too.